### PR TITLE
Update issue transfer documentation

### PR DESCRIPTION
**Pull request by anleac**
### Why:

This reflects the new default behaviour of issues transfer. Labels are now only transferred if they have a match by name in the target repository, and milestones also will be transferred if they match by both name and due-date, so the documentation should reflect this.

This is also reflected today in the dialogue used for transferring:

image

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by cmwilson21** [comment]
👋 @anleac Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ To transfer an open issue to another repository, you must have write access to t
 
 {% endnote %}
 
-When you transfer an issue, comments, labels and assignees are retained. The issue's milestones are not retained. This issue will stay on any user-owned or organization-wide project boards and be removed from any repository project boards. For more information, see "[About project boards](/articles/about-project-boards)."
+When you transfer an issue, comments and assignees are retained. The issue's labels and milestones are also retained if they're present in the target repository, with labels matching by name and milestones matching by both name and due date. This issue will stay on any user-owned or organization-wide project boards and be removed from any repository project boards. For more information, see "[About project boards](/articles/about-project-boards)."
```
Comment (sophietheking) [inline content/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository.md:26]:
```suggestion
When you transfer an issue, comments and assignees are retained. Labels and milestones are also retained if they're present in the target repository, with labels matching by name and milestones matching by both name and due date. This issue will stay on any user-owned or organization-wide project boards and be removed from any repository project boards. For more information, see "[About project boards](/articles/about-project-boards)."
```

**Review by sophietheking — DISMISSED** [review]
Hi @anleac 👋  thanks for opening this PR! Looks good to me, only left one small suggestion for your consideration, but feel free to ignore it if it doesn't make sense to you. 

Please feel free to `@mention` me here and I'll get this merged for you!

**Comment by anleac** [comment]
> Hi @anleac 👋 thanks for opening this PR! Looks good to me, only left one small suggestion for your consideration, but feel free to ignore it if it doesn't make sense to you.
> 
> Please feel free to `@mention` me here and I'll get this merged for you!

thanks for that suggestion @sophietheking, it absolutely makes things cleaner. Committed!

**Review by sophietheking — APPROVED** [review]

**Pull request merged**